python_codes/test_simulated_maps8.py

```diff
@@ -0,0 +1,273 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Sep 27 14:27:26 2018
+7: reprise feb 2020
+Notably we try other pattern of borders (the generic ones) to make propensity maps.
+"""
+import scipy.ndimage
+import scipy.io as sio
+from pylab import *
+import random
+import itertools
+
+import scipy.stats as stats
+from scipy.stats import rv_discrete
+
+import glob
+import os
+import time
+
+import statsmodels.api as sm
+lowess = sm.nonparametric.lowess
+import matplotlib
+
+os.chdir("/home/axel/Bureau/z_python_scripts_copy/")
+import scn
+import ice_mirny3
+import distance_law_human
+import hicstuff as hcs
+
+# Experimental data:
+m=loadtxt("/home/axel/Bureau/YEAST/data_yeast_pds5/ALL_EXPERIMENTS_PDF/AT147_Pds5-AID-noTir1-G1-cdc20-TS_files/txt/MAT_RAW_chr5_AT147_Pds5-AID-noTir1-G1-cdc20-TS_2kb.txt")
+Nreads = sum(m)
+Nrealisations = 10
+
+#imshow(m**0.15,interpolation="none",cmap= 'afmhot_r')
+#colorbar()
+n1 = shape(m)[0]   # size of the chromosome contact map 
+
+# Normalisation:
+matscn1=ice_mirny3.ice_func(m,0)
+matscn1=scn.scn_func(matscn1,0)
+
+
+# Borders from experimental data:
+chrm=5
+b = loadtxt("/home/axel/Bureau/YEAST/data_yeast_pds5/borders_detection/borders_POISSIAN_Dplus_chr"+str(chrm)+"_.txt", dtype="int")
+b = list(b)
+ele_to_remove = [221,224,246,249]
+for e in ele_to_remove :
+    if e in b:
+        b.remove(e)
+
+#plot(b, b, 'o', color="cyan")
+
+# To convert the 1D vetor into matrice object:
+MAT_INDICES = np.zeros(( shape(m)  ))
+compt = 0
+VECT_COMPT = []
+for i, j in itertools.product(range(n1), range(n1)):
+    MAT_INDICES[i,j] = compt
+    VECT_COMPT.append( (i,j) )
+    compt += 1
+#imshow(MAT_INDICES, interpolation ="none")
+
+
+#  Genomic distance law from experiemental data: 
+s = shape(matscn1)[0]
+print(s)
+prob_d = distance_law_human.dist_law(matscn1)
+prob_d =  prob_d / sum(prob_d)
+#plot(prob_d)
+
+# LOWESS Fit:
+z = lowess(prob_d[:,0], range(0, len(prob_d) ) , frac=1./15)
+plot(z[:,0],z[:,1],lw=3)
+prob_d_lowess = z[:,1]
+
+# TADs size distribution:
+list_size=[]
+for i in range(1,len(b) ) :
+    bi = b[i] - b[i-1]
+    list_size.append(bi)
+mean(list_size) 
+std(list_size) 
+min(list_size)
+max(list_size) 
+
+s = shape(matscn1)[0]
+
+# Genomic distance law:
+MAT_GENO = np.zeros(( shape(m)  ))
+for i in range(0,s) :
+    for j in range(0,s) :
+        MAT_GENO[i,j] = prob_d_lowess[ abs(j-i) ]
+ 
+if(len(MAT_GENO[MAT_GENO<0]) >0) :
+    print("attention presence of neg elements")
+    print(len(MAT_GENO[MAT_GENO<0]) )  
+MAT_GENO[MAT_GENO<0]=0       
+
+# Generic pattern:
+RATIO_CIDs = loadtxt("/home/axel/Bureau/tools/chromosight/chromosight/kernels/artificial_template_borders_type1.txt")
```

Comment:
Load chromosight kernels instead (these are the same)

Reply:
Fixed in 786de98cbe6cae7937266e63400c8a2d9a7f3de7, chromosight.kernels is now accessed directly to load kernels